### DISCUSSION TITLE
Update waveform [solved]

### DISCUSSION THREAD

**Discussion by GillesLACAUD** [marcel-licence/esp32_basic_synth#9] [Ideas]
When you change the waveform the result must be immediate. In your code the new wave is active only when you hit a new key

**Comment by marcel-licence** [comment]
The implementation was intentional (I had changing of presets in mind) which should not change immediately. But I fully agree with your statement: we are changing the "preset" / sound setting which is played -> I also want the detune to be changed immediately.

**Comment by marcel-licence** [comment]
The waveform does now change with my latest commit. Thanks for this point =)
